### ISSUE TITLE
Custom templates cause ExpressionChangedAfterItHasBeenCheckedException

### ISSUE THREAD

**Issue by CodeJason** [michaelbromley/ng2-pagination#15]
When I try to create a custom template within the controls, it throws the exception: 

`Expression p.isFirstPage() has changed after it was checked. Previous value: 'false'. Current value: 'true' in [p.isFirstPage()]`

Which I understand is something fixable by using a change detector ref, but efforts so for haven't proved useful. 
I could try and reproduce with plunkr but I thought I'd check if you had tested this with recent versions of angular2?

**Comment by michaelbromley** [comment]
Hi,

What version of this module are you using? The latest is beta.2 and is tested against angular2 beta.7.

Can you show me the code which yields this error?

btw, right now there is no simple way to make a Plunkr demo with this module (as far as I know) because it does not support systemjs. I plan to incorporate a systemjs-compatible build as soon as I get some time to do so.

**Comment by jayhahn** [comment]
Hi, this is the same issue as #12.

Custom Template doesn't work if you place the pagination-control before the actual ngFor elements. @michaelbromley I don't know how to create a plunkr but you can easily reproduce by checking out a quick angular2-seed project and then copy-pasting your example custom-plate to it. Should be able to reproduce in few minutes.

**Comment by michaelbromley** [comment]
There is now a simple [Plunker demo](http://plnkr.co/edit/JVQMPvV8z2brCIzdG3N4?p=preview) you can modify to try to reproduce the error. If you have time please replicate the error so I can investigate.

**Comment by CodeJason** [comment]
Hi, 
Sorry for the long reply, been busy.
I've tried reproducing the issue on plunkr for a little while, but I can't seem to. 
I've looked back in back in the project where I had the problem but I can't remember if it was a simple mistake or wrapping it in a ngIf.
Here's my custom template anyway, I added a service and tried other things but nothing fired the same error message. http://plnkr.co/edit/PFGn07aUISYnPPSU3ktb?p=preview
Thanks.
